### DISCUSSION TITLE
support for dds/dds.hpp that contains all modules?

### DISCUSSION THREAD

**Discussion by jfergusonmi** [eProsima/Fast-DDS#5334] [General]
I am coming from projects that used CDX and RTI. They both provide dds/dds.hpp that contains all important hpp headers and/or all standard symbols and includes. So for example, a hello_world program that sends data could provide just #include dds/dds.hpp instead of the exact 

#include <fastdds/dds/domain/DomainParticipantFactory.hpp>
#include <fastdds/dds/log/Log.hpp>

for example RTI provides in their dds/dds.hpp

#include <dds/core/ddscore.hpp>
#include <dds/domain/ddsdomain.hpp>
#include <dds/topic/ddstopic.hpp>
#include <dds/sub/ddssub.hpp>
#include <dds/pub/ddspub.hpp>

I looked and couldnt find a top-level hpp file for FastDDS. Is there one or would i need to make my own? What would be the main includes needed. I assume something like

#if defined(DDS_PLATFORM_RTI) || defined(DDS_PLATFORM_COREDX)
#include <dds/dds.hpp>
#elif defined(DDS_PLATFORM_FASTDDS)
//#include <fastdds/dds/core/?????> --not sure
#include <fastdds/dds/domain/DomainParticipant.hpp>
#include <fastdds/dds/domain/DomainParticipantFactory.hpp>
#include <fastdds/dds/domain/DomainParticipantListener.hpp>
#include <fastdds/dds/log/Log.hpp>
#include <fastdds/dds/publisher/Publisher.hpp>
#include <fastdds/dds/subscriber/Subscriber.hpp>
#include <fastdds/dds/topic/Topic.hpp>
#include <fastdds/dds/xtypes/utils.hpp>
#endif

**Comment by JesusPoderoso** [comment]
Hi @jfergusonmi,
at eProsima, we enhance to follow the "include-what-you-use" philosophy in order to avoid unncecessary cost in build time.
That is the reason why this convience header is not part of the Fast DDS library.  
You can take the [hello world example](https://github.com/eProsima/Fast-DDS/tree/master/examples/cpp/hello_world) as a reference of the header files you would need to include for a "basic" Fast DDS deployment.

The following code snippet is a mix of header files included in the example, hope it helps:

```cpp
#include <fastdds/dds/core/status/SubscriptionMatchedStatus.hpp>
#include <fastdds/dds/domain/DomainParticipant.hpp>
#include <fastdds/dds/domain/DomainParticipantFactory.hpp>
#include <fastdds/dds/log/Log.hpp>
#include <fastdds/dds/topic/TypeSupport.hpp>
#include <fastdds/dds/publisher/DataWriter.hpp>
#include <fastdds/dds/publisher/DataWriterListener.hpp>
#include <fastdds/dds/publisher/Publisher.hpp>
#include <fastdds/dds/publisher/qos/DataWriterQos.hpp>
#include <fastdds/dds/publisher/qos/PublisherQos.hpp>
#include <fastdds/dds/subscriber/DataReader.hpp>
#include <fastdds/dds/subscriber/DataReaderListener.hpp>
#include <fastdds/dds/subscriber/qos/DataReaderQos.hpp>
#include <fastdds/dds/subscriber/qos/SubscriberQos.hpp>
#include <fastdds/dds/subscriber/SampleInfo.hpp>
#include <fastdds/dds/subscriber/Subscriber.hpp>
```
